pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -0,0 +1,149 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package cluster
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	as_util "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	log "github.com/cihub/seelog"
+	yaml "gopkg.in/yaml.v2"
+	"time"
+)
+
+// Covers the Controle pannel service check and the in memory pod metadata.
+
+const kubernetesApiServerCheck = "kubernetes_apiserver"
+
+const (
+	KubeControlPaneCheck string = "kubernetes_APIServer.controlePlaneUp"
+)
+
+type KubeASConfig struct {
+	//CollectContainerSize bool               `yaml:"collect_container_size"`
+	Tags []string `yaml:"tags"`
+	//CollectEvent         bool               `yaml:"collect_events"`
+	//FilteredEventType    []string           `yaml:"filtered_event_types"`
+}
+
+// KubeASCheck grabs docker metrics
+type KubeASCheck struct {
+	lastWarnings []error
+	instance     *KubeASConfig
+	//lastEventTime  time.Time
+	KubeApiServerHostname string
+}
+
+func (k *KubeASCheck) String() string {
+	return kubernetesApiServerCheck
+}
+
+func (c *KubeASConfig) Parse(data []byte) error {
```

Comment:
I don't think Parse is inside the check interface. It can be private (lowercase).